server/hydrogen-render/1-render-hydrogen-to-string.js

```diff
@@ -24,9 +24,30 @@ async function renderHydrogenToString(options) {
     // we receive the SSR results.
     const child = fork(require.resolve('./2-render-hydrogen-to-string-fork-script'), [], {
       signal,
+      // Make `child.stderr` and `child.stdout` available
+      silent: true,
       //cwd: process.cwd(),
     });
 
+    // Since we have to use the `silent` option for the `stderr` stuff below, we
+    // should also print out the `stdout` to our main console.
+    child.stdout.on('data', function (data) {
+      console.log('Child printed something to stdout:', String(data));
+    });
+
+    // It's possible for the child to exit and print an error to the `stderr`
+    // without it actually bubbling up to us in the parent. As far as I can
+    // tell, the `try`/`catch` in `2-render-hydrogen-to-string-fork-script.js`
+    // just isn't catching the error (doesn't make any sense).
+    child.stderr.on('data', function (data) {
+      const stubError = new Error();
+      stubError.name = 'PrintFromStdErr';
+      stubError.message = '';
+      stubError.stack = String(data);
+      const childError = new RethrownError(`Child process printed something to stderr:`, stubError);
+      childErrors.push(childError);
+    });
```

Comment:
It would be really nice to figure out why the error doesn't bubble up to us in the parent even though it shows that the stack trace shows it's within the `try/catch` that would do this. Plus it doesn't appear on `window.addEventListener('error', ...)` or window.addEventListener('unhandledrejection', ...). I've spent a good chunk of time investigating but I don't see how to capture this error. It just appears in the console `stderr`.

```
TypeError: Cannot read properties of undefined (reading 'storeNames')
    at TimelineReader.readById (C:\Users\MLM\Documents\GitHub\element\hydrogen-web\target\lib-build\hydrogen.cjs.js:12483:33)
    at Timeline._getEventFromStorage (C:\Users\MLM\Documents\GitHub\element\hydrogen-web\target\lib-build\hydrogen.cjs.js:12762:46)
    at Timeline._loadContextEntryNotInTimeline (C:\Users\MLM\Documents\GitHub\element\hydrogen-web\target\lib-build\hydrogen.cjs.js:12747:35)
    at Timeline._loadContextEntriesWhereNeeded (C:\Users\MLM\Documents\GitHub\element\hydrogen-web\target\lib-build\hydrogen.cjs.js:12741:14)
    at Timeline.addEntries (C:\Users\MLM\Documents\GitHub\element\hydrogen-web\target\lib-build\hydrogen.cjs.js:12699:10)
    at mountHydrogen (4-hydrogen-vm-render-script.js:204:12)
    at 4-hydrogen-vm-render-script.js:353:1
    at Script.runInContext (node:vm:139:12)
    at _renderHydrogenToStringUnsafe (C:\Users\MLM\Documents\GitHub\element\matrix-public-archive\server\hydrogen-render\3-render-hydrogen-to-string-unsafe.js:102:41)
    at async process.<anonymous> (C:\Users\MLM\Documents\GitHub\element\matrix-public-archive\server\hydrogen-render\2-render-hydrogen-to-string-fork-script.js:18:27)
```

Reply:
Maybe it's just being logged with `console.error` although seems a bit strange for a `TypeError` on it's own to do that.

Reply:
See https://github.com/matrix-org/matrix-public-archive/pull/43#discussion_r927172956

We can catch it with `rocess.on('uncaughtException',` ✅